process-services/1.11/develop/dev-ext.md

```diff
@@ -0,0 +1,1922 @@
+
+## Custom reports
+
+There are a number of out-of-the-box reports in the Analytics app, which can be augmented with your own custom reports.
+
+Custom reports have full access to the Elasticsearch indexes generated by Process Services when it is enabled.
+
+See [Event processing for analytics]({% link process-services/1.11/config/external.md %}#event-processing-for-analytics) for details on how to configure events to be sent to Elasticsearch.
+
+The following section assumes that you have a reasonable understanding of what Elasticsearch is and an understanding of indexes, types and type mappings. The [Elasticsearch Definitive Guide](https://www.elastic.co/guide/en/elasticsearch/guide/1.x/index.html){:target="_blank"} is a great learning resource if you are new to the engine and there is also a [Reference Guide](https://www.elastic.co/guide/en/elasticsearch/reference/1.7/index.html){:target="_blank"} which you should find helpful to refer to as you start using it directly yourself.
+
+### Implementing custom reports
+
+Assuming that you have started to see some data show up in the ElasticSearch store and therefore in the out-of-the-box reports, and you have used the Sense tool or cURL to develop some custom search queries of your own, you are ready to start implementing the custom Spring bean required in order to plug the report into the Process Services UI.
+
+1. Basic concepts
+
+    A custom report is a custom section available in the Analytics app and also within each published app, which shows one or more custom reports.
+
+    Each report is implemented by a Spring bean which is responsible for two things:
+
+    1. Perform an ElasticSearch search query using the Java client API.
+
+    2. Convert the search results (hits or aggregations) into chart or table data and add this to the response.
+
+    The UI will automatically display the correct widgets based on the data that your bean sends.
+
+2. Bean implementation
+
+    Your Spring bean will be discovered automatically via annotations but must be placed under the package `com.activiti.service.reporting`. Since this package is used for the out-of-the-box reports it is recommended that custom reports use the sub-package such as `com.activiti.service.reporting.custom`.
+
+    The overall structure of the class will be as follows, for the full source please see the web link at the end of this section.
+
+    ```java
+    package com.activiti.service.reporting.custom;
+    
+    import com.activiti.domain.reporting.ParametersDefinition;
+    import com.activiti.domain.reporting.ReportDataRepresentation;
+    import com.activiti.service.api.UserCache;
+    import com.activiti.service.reporting.AbstractReportGenerator;
+    import org.activiti.engine.ProcessEngine;
+    import org.elasticsearch.client.Client;
+    import org.springframework.stereotype.Component;
+    
+    import java.util.Map;
+    
+    @Component(CustomVariablesReportGenerator.ID)
```

Comment:
Needs review. Probably code needs to be updated

Reply:
Spoken Bassam will tackle this with another JIRA after release. more info needs gathering from a dev.